pkg/network/link.go

```
@@ -76,3 +78,49 @@ func getLinkIPAddresses(link netlink.Link, scope int) ([]net.IP, error) {
 	}
 	return ips, nil
 }
+
+// GetLinkIPAddrForIP gets the netlink.Addr for the given link name and IP address.
+func GetLinkIPAddrForIP(name string, ip net.IP) (*netlink.Addr, error) {
+	link, err := netlink.LinkByName(name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get link %s: %w", name, err)
+	}
+	addrs, err := netlink.AddrList(link, familyAll)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list addresses of link %s: %w", link.Attrs().Name, err)
+	}
+	for _, addr := range addrs {
+		if addr.IP.Equal(ip) {
+			return &addr, nil
+		}
+	}
+	return nil, nil
```

Should we return an error, when the IP is not found?